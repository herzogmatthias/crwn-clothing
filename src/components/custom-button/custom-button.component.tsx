import * as React from "react";
import "./custom-button.styles.scss";

export interface ICustomButtonProps {
  children: string;
  onClick?(): void;
  type?: "button" | "submit" | "reset" | undefined;
}

export function CustomButton({ children, ...otherProps }: ICustomButtonProps) {
  return (
    <button className="custom-button" {...otherProps}>
      {children}
    </button>
  );
}
