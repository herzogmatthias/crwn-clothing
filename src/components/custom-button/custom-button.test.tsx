import React from "react";
import { shallow } from "enzyme";
import { CustomButton } from "./custom-button.component";
describe("NON CONNECTED CUSTOM-BUTTON TESTS", () => {
  test("render Custom Button", () => {
    expect(shallow(<CustomButton />)).toMatchSnapshot();
  });
});
