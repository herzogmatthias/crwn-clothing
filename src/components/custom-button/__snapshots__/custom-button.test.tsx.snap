// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NON CONNECTED CUSTOM-BUTTON TESTS render Custom Button 1`] = `
<styled.button
  inverted={false}
  isGoogleSignIn={false}
/>
`;
